### PR TITLE
DOCS-83 Microsoft 365

### PR DESCRIPTION
**Pull request by reidstidolph**
Initial draft doc for how to use the built in Microsoft 365 module.

### CODE REVIEW

**Comment by reidstidolph** [comment]
Looks like the build is failing due to the `cli_reference.md` doc ( https://docs.128technology.com/docs/cli_reference/#show-entitlement ) having a broken link to https://www.128technology.com/entitlement . If we can figure out where the new URL is on our main site for this entitlement information, I can fix the link.

File: docs/howto_ms365.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+---
+title: Microsoft 365
+sidebar_label: Microsoft 365
+---
+
+128T optimizes Microsoft 365 sessions, allowing you to easily configure the associated services to be delivered using the reccomended [network connectivity principles](https://docs.microsoft.com/en-us/office365/enterprise/office-365-network-connectivity-principles). It uses an [AppID module](/docs/concepts_appid/#appid-using-modules) for automatic discovery of Microsoft 365 endpoints, and simple service definition.
```
Comment (MichaelBaj) [inline docs/howto_ms365.md:6]: We utilize a flat directory structure.  `/docs/concepts_appid/#appid-using-modules` is incorrect.  It should instead be `concepts_appid.md#appid-using-modules`

- [ ] Correct internal link

File: docs/howto_ms365.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+:::note
```
Comment (MichaelBaj) [inline docs/howto_ms365.md:8]: `info` might be more appropriate.

File: docs/howto_ms365.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        exit
+    exit
+exit
+```
+
+### Service Policies
+With your services for Microsoft 365 added, the final step is to configure the services for which tenants can access Microsoft 365, security policies, routing, and optional service-policy.
+
+#### Access
+Define which of your [network tenants](/docs/config_tenants/#modeling-your-network-tenancy) are allowed to access Microsoft 365 services by adding them to the [service access policies](/docs/config_reference_guide/#access-policy-service).
```
Comment (MichaelBaj) [inline docs/howto_ms365.md:103]: internal link should be `config_tenants.md#modeling-your-network-tenancy`

- [ ] Fix internal link

Comment (MichaelBaj) [inline docs/howto_ms365.md:103]: internal link should be `config_reference_guide.md#access-policy-service`

- [ ] Fix internal link

File: docs/howto_ms365.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+#### Security
+Configure the [security policy](/docs/config_reference_guide/#security) for the Microsoft 365 services.
```
Comment (MichaelBaj) [inline docs/howto_ms365.md:106]: Internal link should be `config_reference_guide.md#security`

- [ ] Fix internal link

File: docs/howto_ms365.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+:::tip
+Microsoft 365 services are designed for direct Internet transport, with robust security and encryption built in to each application. As such, a security policy that disables router encryption is typically preferred.
+:::
+
+#### Routing
+Give the Microsoft 365 services appropriate [service routes](/docs/concepts_glossary/#service-routes) in your overall topology to designate possible routes.
```
Comment (MichaelBaj) [inline docs/howto_ms365.md:113]: Internal link should be `config_reference_guide.md#security`

- [ ] Fix internal link

File: docs/howto_ms365.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+:::tip
+In broadband connected branch router scenarios, the optimal route for M365 services will typically be directly out local broadband links. In this case, disabling the sharing of service-routes may be desirable in the service configuration.
+:::
+
+#### Service Policy
+By using [service policy](/docs/bcp_service_and_service_policy_design/#service-policy) associated with your Microsoft 365 services, you can set how your network should treat sessions. This includes QoS, path selection, and failover behavior for Microsoft 365.
```
Comment (MichaelBaj) [inline docs/howto_ms365.md:120]: Internal link should be `bcp_service_and_service_policy_design.md#service-policy`

- [ ] Fix internal link

Comment (reidstidolph) [reply docs/howto_ms365.md]: fixed.

File: docs/howto_ms365.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+Once module AppID learning mode has been enabled, the following application names will be automatically discovered (see [Verifying Operation](#verifying-operation) to confirm successful discovery): 
+
+| Application Name  | Description                                   |
+| ----------------- | --------------------------------------------- |
+| `O365-Common`     | Microsoft 365 Common and Office Online        |
+| `O365-Exchange`   | Exchange Online                               |
+| `O365-SharePoint` | SharePoint Online and OneDrive for Business   |
+| `O365-Skype`      | Skype for Business Online and Microsoft Teams |
+
+Create one or more services configured for the module discovered application names.
```
Comment (agrawalkaushik) [inline docs/howto_ms365.md:57]: Just FYI, the O365 app-id module has a notion of service-mapping JSON that can be put in place. I've not used it but I could try it - we should update the document with notes on that at some point.

Comment (reidstidolph) [reply docs/howto_ms365.md:57]: Oh, interesting. Yeah perhaps we can update this at some point with that additional info.

**Comment by reidstidolph** [comment]
Merging M365 module doc.

**Pull request merged**